### ISSUE TITLE
The `grafana4_user.present` state called with test=True Creates Users in salt-ssh-2017.7.1

### ISSUE THREAD

**Issue by msummers42** [saltstack/salt#43675]
### Description of Issue/Question
When using salt-ssh-2017.7.1 and the `grafana4_user.present` state module with test=True, like this `salt-ssh grafanabox state.sls grafana.grafana_users test=True` the state creates users instead of testing for change!

### Setup
In file `grafana/grafana_users.sls`
Our user looks like this from pillar:
```
fbarbazzz:
  email: foo@example.com
  fullname: Foo Barbazzz
  team: ops
 ```
Then in another pillar we have the profile information. None of this is at issue.
```
ensure {{ user }} in grafana:
  grafana4_user.present:
    - name: {{ args['email'] }}@syapse.com
    - email: {{ args['email'] }}@syapse.com
    - fullname: {{ args['fullname'] }}
    - is_admin: {{ True if args['team'] == "ops" else False }}
    - password: {{ salt.test.rand_str(size=32) }}
    - profile:
        grafana_url: {{ pillar['grafana']['grafana_url'] }}
        grafana_user: {{ pillar['grafana']['admin_user'] }}
        grafana_password: {{ pillar['grafana']['admin_password'] }}
        grafana_timeout: 5
```

### Steps to Reproduce Issue
The user is created in grafana, state output reflects that as well as grafana itself.

### Versions Report
```
Salt Version:
           Salt: 2017.7.1
 
Dependency Versions:
           cffi: 1.10.0
       cherrypy: Not Installed
       dateutil: 2.6.0
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.5
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: Not Installed
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.2
            ZMQ: Not Installed
 
System Versions:
           dist: gentoo 2.3 
         locale: UTF-8
        machine: x86_64
        release: 4.9.51
         system: Linux
        version: Gentoo Base System 2.3
```

**Comment by msummers42** [comment]
This issue is present with the grafana4_orgs module as well. `test=True` is mishandled. It's clear from viewing the API calls to the grafana node.
```
# Does the org exist?
"GET /api/orgs/name/Engineering HTTP/1.1" 404
# No, and it should probably stop here, but it does not.
# It POSTs to create it
"POST /api/orgs HTTP/1.1" 200
# Then it tests again for existence.
GET /api/orgs/name/OrgNameUnderTest HTTP/1.1" 200
```

**Comment by gtmanfred** [comment]
It looks like neither of the modules specified actually check `__opts__['test'] == True`

@bbinet it looks like you wrote these, would you mind taking a look and adding test=True support?

Thanks,
Daniel

**Comment by bbinet** [comment]
@gtmanfred You're right, the `__opts__['test']` is not supported yet.
I don't have time right now to implement it, but as a quick fix, we could just return a message when  `__opts__['test'] == True` to say the "test" feature is not implemented yet and to avoid the grafana instance to be actually updated.

**Comment by bbinet** [comment]
Also note that the grafana module should also be updated to avoid displaying false positive diffs with newest grafana version.

**Comment by bbinet** [comment]
This should be fixed when PR #48671 is merged.

**Comment by gtmanfred** [comment]
awesome, thanks for the pr @bbinet
